src/cpp/r/RSexp.cpp

```diff
@@ -172,24 +196,58 @@ SEXP findVar(const std::string& name, const std::string& ns)
    
    return findVar(name, env);
 }
-  
+
 SEXP findFunction(const std::string& name, const std::string& ns) 
 {
+   r::sexp::Protect protect;
```

Comment:
Okay, sounds good. Note that it is possible to avoid the R error being thrown via executeSafely (not sure if you've seen that before -- it's our wrapper for R_ToplevelExec)